Sources/MapboxMaps/Foundation/CredentialsManager.swift

```diff
@@ -0,0 +1,100 @@
+import Foundation
+
+/// Convenience class that holds MapboxMaps related secrets.
+///
+/// It's possible to create `CredentialsManager` instances as you need them,
+/// however it's convenient to use the default object (`default`).
+///
+/// For example, we recommend that the Mapbox access token be set in
+/// `application(_:didFinishLaunchingWithOptions:)` rather than relying on the
+/// value in your application's Info.plist:
+///
+///     ```
+///     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+///         // Override point for customization after application launch.
+///         CredentialsManager.default.accessToken = "overridden-access-token"
+///         return true
+///     }
+///     ```
+public class CredentialsManager {
+
+    /// Access token
+    public var accessToken: String
+
+    /// Default instance
+    ///
+    /// This shared instance is used by the default initializers
+    /// for `ResourceOptions` and `MapInitOptions`.
+    ///
+    /// The application's Info.plist will be searched for a valid access token
+    /// under the key `MBXAccessToken`
+    ///
+    /// A valid access token must be provided or found.
+    public static var `default` = CredentialsManager(internal: nil)
+
+    /// Initializes a CredentialsManager with an access token.
+    ///
+    /// Use the shared `CredentialsManager.default` to set a globally shared
+    /// access token.
+    ///
+    /// - Parameter accessToken: access token
+    public convenience init(accessToken: String) {
```

Comment:
The fact that this takes a non-optional String means that if I change the value of `default`, I'll never be able to set it back to a CredentialsManager with nil for the access token.

Reply:
Default should be a `private(set)` - good catch.

Reply:
Ticketed.